pilot/pkg/model/push_context.go

```diff
@@ -137,7 +137,39 @@ type XDSUpdater interface {
 	// ConfigUpdate is called to notify the XDS server of config updates and request a push.
 	// The requests may be collapsed and throttled.
 	// This replaces the 'cache invalidation' model.
-	ConfigUpdate(full bool)
+	ConfigUpdate(req UpdateReq)
+}
+
+// UpdateReq defines a request to update proxies
+type UpdateReq struct {
+	// Full determines whether a full push is required or not. If set to false, only endpoints will be sent.
+	Full bool
+
+	// TargetNamespaces contains a list of namespaces that were changed in the update.
+	// This is used as an optimization to avoid unnecessary pushes to proxies that are scoped with a Sidecar.
+	// Currently, this will only scope EDS updates, as config updates are more complicated.
+	// If this is empty, then proxies in all namespaces will get an update
+	// If this is present, then only proxies that import this namespace will get an update
+	TargetNamespaces map[string]struct{}
+}
+
+// Merge two update requests together
+func (first *UpdateReq) Merge(other *UpdateReq) *UpdateReq {
```

Comment:
Can we just make this structure immutable and instead have `Merge` return a new instance? Also, you return by value rather than reference?